### DISCUSSION TITLE
Setting x-kubernetes-embedded-resource: false in CRD

### DISCUSSION THREAD

**Discussion by kmathewmk** [kubernetes-sigs/kubebuilder#3260] [Support]
How can I use 
`//+kubebuilder:validation:EmbeddedResource`
mentioned in 
https://book.kubebuilder.io/reference/markers/crd-validation.html
to set 
`x-kubernetes-embedded-resource: false`
in the CRD yaml to support resources without fields apiVersion/kind/metadata. 

The field annonation does not seem to support a boolean as a value.

The objective is to pass through a CR objects (configuration data) without fields like kind/apiVersion/...

**Comment by camilamacedo86** [comment]
The marker `//+kubebuilder:validation:EmbeddedResource` is used to set `x-kubernetes-embedded-resource: true` for a field in the CRD. This will ensure that the field contains a valid embedded Kubernetes runtime.Object, meaning it should have `kind`, `apiVersion`, and `metadata` fields. However, If you don't annotate the field with the `EmbeddedResource` marker, it should assume this behavior.

Seems that to do what you are looking for your can avoid to use `//+kubebuilder:validation:EmbeddedResource` marker for the field in your Go type definitions.

For instance:

\```go
type MyCustomSpec struct {
    Configuration SomeConfig `json:"configuration"`
}

type SomeConfig struct {
    ConfigField1 string `json:"configField1"`
    ConfigField2 int    `json:"configField2"`
}
\```

In the above struct, `SomeConfig` does not have `kind`, `apiVersion`, or `metadata`, and we're not using the `EmbeddedResource` marker, so it won't enforce the presence of those fields in the CRD.

However, if that does not answered your question please raise an issue agains controller-tools project (https://github.com/kubernetes-sigs/controller-tools) it is the project responsible for the markers so you might find a better help.

I hope that helps you out.
Closing this one. 
However, feel free to re-open if you see that is required,

**Comment by kmathewmk** [comment]
I no longer have the details of the issue I was facing even when `//+kubebuilder:validation:EmbeddedResource` was not specified. I shall reopen later if necessary.
